src/sentry/workflow_engine/endpoints/serializers/group_open_period_serializer.py

```diff
@@ -41,12 +41,20 @@ def serialize(
 @register(GroupOpenPeriod)
 class GroupOpenPeriodSerializer(Serializer):
     def get_attrs(self, item_list, user, **kwargs):
+        query_start = kwargs.get("query_start")
+        query_end = kwargs.get("query_end")
         result: defaultdict[GroupOpenPeriod, dict[str, list[GroupOpenPeriodActivityResponse]]] = (
             defaultdict(dict)
         )
         activities = GroupOpenPeriodActivity.objects.filter(
             group_open_period__in=item_list
-        ).order_by("id")
+        ).order_by("date_added")
+
+        if query_start:
+            activities = activities.filter(date_added__gte=query_start)
```

Comment:
If I'm reading this correctly, this seems like it will not return an activity that was created before the time range, but ends during/after the time range. We will need that to be returned in order to populate the chart correctly

Like this:

```
    [ <----- time range -----> ]
[activity_1][activity_2][activity_3]
```